internal/executor/instance/container/container.go

```diff
@@ -58,12 +48,17 @@ func (inst *Instance) Run(ctx context.Context, config *runconfig.RunConfig) (err
 			return
 		}
 
-		cleanupErr := agentVolume.Close(config.ContainerBackend)
+		containerBackend, creationError := config.GetContainerBackend()
```

Comment:
This needs to use a fast-fast pattern otherwise we'll hit NPE in the `agentVolume.Close()` below.